backend/src/Faryma.Composer.Core/Features/ReviewFeature/ReviewService.cs

```diff
@@ -10,21 +10,61 @@ public sealed class ReviewService(
         UnitOfWork ofw,
         OrderQueueService orderQueueService)
     {
-        public async Task CompleteReview(CompleteReviewCommand command)
+        public async Task<Review> CompleteReview(CompleteReviewCommand command)
         {
             ReviewOrder order = await ofw.ReviewOrderRepository.Find(command.ReviewOrderId)
                 ?? throw new ReviewException($"Заказ разбора трека Id: {command.ReviewOrderId}, не существует");
 
-            if (order.Status != ReviewOrderStatus.Pending)
+            if (order.Status != ReviewOrderStatus.InProgress)
             {
-                throw new ReviewException($"Невозможно поднять заказ в статусе '{order.Status}'");
+                throw new ReviewException("Невозможно оценить заказ т.к. его статус не в процессе");
             }
 
-            // ...
+            Review review = await CreateReview(command, order);
+
+            order.Status = ReviewOrderStatus.Completed;
+            order.Review = review;
 
             await ofw.SaveChangesAsync();
 
             await orderQueueService.CompleteReview(order);
+
+            return review;
+        }
+
+        private async Task<Review> CreateReview(CompleteReviewCommand command, ReviewOrder order)
+        {
+            Track track;
+
+            if (order.TrackId.HasValue)
+            {
+                track = await ofw.TrackRepository.Find(order.TrackId.Value)
+                    ?? throw new ReviewException($"Трека {order.TrackId.Value} не существует");
+            }
+            else if (!string.IsNullOrWhiteSpace(order.TrackUrl))
+            {
+                UserNickname userNickname = order.UserNicknames.First();
+                track = await ofw.TrackRepository.GetOrCreateByUrl(userNickname, order.TrackUrl);
+            }
+            else
+            {
+                throw new ReviewException($"У заказа {order.Id} нет трека");
+            }
+
+            Review trackReview = new()
+            {
+                Comment = command.Comment,
+                CompletedAt = DateTime.UtcNow,
```

Comment:
https://github.com/farymacomposer/webapp/wiki/%D0%A0%D1%83%D0%BA%D0%BE%D0%B2%D0%BE%D0%B4%D1%81%D1%82%D0%B2%D0%BE-%D0%BF%D0%BE-%D0%BA%D0%BE%D0%B4%E2%80%90%D1%80%D0%B5%D0%B2%D1%8C%D1%8E-C%23-(%D1%87%D0%B0%D1%81%D1%82%D1%8B%D0%B5-%D0%BE%D1%88%D0%B8%D0%B1%D0%BA%D0%B8)#%D0%B8%D1%81%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5-datetimeutcnow-datetimenow-%D0%B8%D0%BB%D0%B8-datetimetoday-%D0%B2-%D0%BC%D0%B5%D1%81%D1%82%D0%B0%D1%85-%D0%B3%D0%B4%D0%B5-%D1%8D%D1%82%D0%BE-%D0%BC%D0%BE%D0%B6%D0%B5%D1%82-%D0%BF%D1%80%D0%B8%D0%B2%D0%B5%D1%81%D1%82%D0%B8-%D0%BA-%D0%BD%D0%B5%D1%81%D0%BE%D0%B3%D0%BB%D0%B0%D1%81%D0%BE%D0%B2%D0%B0%D0%BD%D0%BD%D0%BE%D1%81%D1%82%D0%B8-%D0%B4%D0%B0%D0%BD%D0%BD%D1%8B%D1%85